lib/p2p/Peer.ts

```diff
@@ -122,14 +127,15 @@ class Peer extends EventEmitter {
    * and emit the `open` event if everything succeeds. Throw an error on unexpected handshake data.
    * @param handshakeData our handshake data to send to the peer
    * @param nodePubKey the expected nodePubKey of the node we are opening a connection with
+   * @param retryConnecting whether to retry to connect upon failure
    */
-  public open = async (handshakeData: HandshakeState, nodePubKey?: string): Promise<void> => {
+  public open = async (handshakeData: HandshakeState, nodePubKey?: string, retryConnecting?: boolean): Promise<void> => {
```

Comment:
I think it's clear enough to just call this new param `retry` which is also consistent with how this flag is named elsewhere.

Edit: Actually after looking through more changes it looks like `retryConnecting` is actually used in most places. Personally I still prefer `retry` but if you want to keep `retryConnecting` then I'd use that same naming in the `initConnection` method.

Reply:
I called it `retryConnecting` in all places except in `initConnection` because there it's applied on the connection context. I can change it to have the same name there as well.

Reply:
To me `retry` even in the other contexts makes enough sense, but it's fine to leave as is. @michael1011 any thoughts on the naming here?